cypress/integration/page.spec.js

```diff
@@ -1,148 +1,1259 @@
+  describe('Page by default', () => {
+    describe('', () => {
+      beforeEach(() => {
+        page.mockUsers();
+        cy.visit('/');
+      });
+
+      it('should have a UserSelector', () => {
+        userSelector.el().should('exist');
+      });
+
+      it('should show NoSelectedUser message', () => {
+        page.noSelectedUserMessage().should('exist');
+      });
+
+      it('should not show posts', () => {
+        page.postsList().should('not.exist');
+        page.posts().should('not.exist');
+      });
+
+      it('should not show posts error message', () => {
+        page.postsLoadingError().should('not.exist');
+      });
+
+      it('should not show no posts yet message', () => {
+        page.noPostsYetMessage().should('not.exist');
+      });
+
+      it('should not show posts loader', () => {
+        page.postsLoader().should('not.exist');
+      });
+    });
+
+    describe('', () => {
+      it('should request users from API', () => {
+        page.spyOn('**/users', 'users');
+
+        cy.visit('/');
+
+        cy.get('@users').should('have.been.called');
+      });
+
+      it('should request users only once', () => {
+        page.spyOn('**/users', 'users');
+
+        cy.visit('/');
+        cy.wait(1000);
+
+        cy.get('@users').should('have.callCount', 1);
+      });
+
+      it('should not request posts from API', () => {
+        page.mockUsers();
+        page.spyOn('**/posts**', 'posts');
+
+        cy.visit('/');
+        cy.waitFor('@usersRequest');
+        cy.wait(500);
+
+        cy.get('@posts').should('not.be.called');
+      });
+
+      it('should not request comments from API', () => {
+        page.mockUsers();
+        page.mockUser1Posts();
+        page.spyOn('**/comments**', 'comments');
+
+        cy.visit('/');
+        cy.waitFor('@usersRequest');
+        cy.wait(500);
+
+        cy.get('@comments').should('not.be.called');
+      });
+    });
+  })
+
+  describe('UserSelector', () => {
+    const { el, button, users, selectedUser } = userSelector;
+
+    describe('', () => {
+      it('should have all the loaded users', () => {
+        page.mockUsers();
+        cy.visit('/');
+
+        users().should('have.length', 10);
+      });
+
+      it('should not have users hardcoded', () => {
+        cy.intercept('**/users', { fixture: 'someUsers' })
+        cy.visit('/');
+
+        users().should('have.length', 3);
+      });
+
+      it('should not have users before they are loaded', () => {
+        cy.clock();
+        page.mockUsers();
+        cy.visit('/');
+        users().should('have.length', 0);
+
+        cy.tick(1000);
+        page.waitForRequest('@usersRequest');
+        users().should('have.length', 10);
+      });
+    });
+
+    describe('after users are loaded', () => {
+      beforeEach(() => {
+        page.mockUsers();
+        cy.visit('/');
+        page.waitForRequest('@usersRequest');
+      });
+
+      it('should should empty text by default', () => {
```

Comment:
There is a typo in the test description: 'should should empty text by default'. It should be 'should show empty text by default'.